src/lib/optimizer/strategy/projection_combination_rule.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class ProjectionNode;
+
+/**
+ * This optimizer rule combines consecutive ProjectionNodes if they all have exactly one child/parent
+ * (Note: The node furthest up in the tree, the "root" of a ProjectionNode sequence,
+ * can have multiple parents; the one furthest down in the sequence can have multiple children).
```

Comment:
ProjectionNodes shouldn't have multiple children.

Reply:
Thanks for pointing that out (and also the comments further down), I've adjusted the rule accordingly! :+1: